injector.c

```diff
@@ -51,29 +53,45 @@ void injector_init(void) {
 }
 
 void injector_close(void) {
-    WHITE_LED_OFF();
-    // turn everything off for a bit to avoid shoot-through
-    LATB4 = 0;
-    LATB5 = 0;
-    LATB2 = 0;
-    LATB3 = 0;
-
-    __delay_us(200);
-    LATB4 = 1;  // high side A -> high
-    LATB3 = 1;  // low side B -> high
+    // check for if first call to injector_close is when the valve is closed
+    if (current_valve_state == VALVE_CLOSED) {
+        // injector is already closed so nothing should happen
+        return;
+    } else {
+        // injector will be closed here
+        WHITE_LED_OFF();
+        // turn everything off for a bit to avoid shoot-through
+        LATB4 = 0;
+        LATB5 = 0;
+        LATB2 = 0;
+        LATB3 = 0;
+        // delay
```

Comment:
Style point: usually it's considered bad form to leave an "obvious" comment since it doesn't add information about what's happening.

Reply:
So should I remove all of the comments?

Reply:
Just the ones I specificied.